checkpoint/recovery.go

```diff
@@ -216,65 +219,149 @@ type recoverydata struct {
 	atxs     []*atxs.CheckpointAtx
 }
 
-type ownAtxData struct {
-	atx      *types.VerifiedActivationTx
-	preserve []*types.VerifiedActivationTx
-	proof    []byte
-}
-
 func preserveOwnData(
+	logger log.Log,
 	db *sql.Database,
 	cfg *RecoverConfig,
 	nodeID types.NodeID,
 	data *recoverydata,
-) (ownAtxData, error) {
-	var own ownAtxData
+) ([]*types.VerifiedActivationTx, [][]byte, error) {
 	if !cfg.PreserveOwnAtx {
-		return own, nil
+		return nil, nil, nil
 	}
 	atxid, err := atxs.GetLastIDByNodeID(db, nodeID)
 	if err != nil && !errors.Is(err, sql.ErrNotFound) {
-		return own, fmt.Errorf("query own last atx id: %w", err)
+		return nil, nil, fmt.Errorf("query own last atx id: %w", err)
 	}
+	var seed types.ATXID
+	var own bool
 	if atxid == types.EmptyATXID {
-		return own, nil
+		// check if any atx is being built
+		if cfg.PostDataDir == "" {
+			return nil, nil, nil
+		}
+		m, err := initialization.LoadMetadata(cfg.PostDataDir)
```

Comment:
do you want to check for positioning atx in nipost challenge? i think it becomes even more error prone , but technically same problem exists there

Reply:
added. thanks